sumpy/e2p.py

```diff
@@ -183,16 +174,17 @@ def get_kernel(self):
                     lp.ValueArg("nsrc_level_boxes,naligned_boxes", np.int32),
                     lp.ValueArg("src_base_ibox", np.int32),
                     lp.ValueArg("ntargets", np.int32),
-                    "..."
+                    ...
                 ] + [arg.loopy_arg for arg in self.expansion.get_args()],
                 name=self.name,
                 assumptions="ntgt_boxes>=1",
                 silenced_warnings="write_race(write_result*)",
-                default_offset=lp.auto,
-                fixed_parameters=dict(dim=self.dim, nresults=len(result_names)),
-                lang_version=MOST_RECENT_LANGUAGE_VERSION)
+                )
 
+        loopy_knl = lp.fix_parameters(loopy_knl,
+            dim=self.dim, nresults=len(result_names))
```

Comment:
Wouldn't it be better for all the checks that the fixing of parameters happens before the checks happen?
https://github.com/inducer/loopy/blob/a4035dd50189e4b4141ef1e25be094e0e72e0289/loopy/kernel/creation.py#L2522

Reply:
Yeah, definitely. I put this outside because the interface for `make_loopy_program` didn't have it, bu it's probably better to add it to that (loopy is already throwing all sorts of warnings because of this)

Thanks for the push!

Reply:
Put all of them back in 2332fd5.